tsc/usb/endpoint.ts

```diff
@@ -117,6 +118,9 @@ export class InEndpoint extends Endpoint {
                     this.pollTransfers = [];
                     this.pollActive = false;
                     this.emit('end');
+                    if (callback) {
+                        callback(error, buffer, actualLength);
```

Comment:
I wonder if we should check for cancelled here and set error to be undefined if it was cancelled.

e.g.
```typescript
if (error.errno === LIBUSB_TRANSFER_CANCELLED) {
    error = undefined;
}
```

Reply:
I'm not positive, however at least in my current workflow, it actually helps to know that it was explicitly cancelled vs some other type of error (for ignoring purposes downstream)

Reply:
Hmm, I see your point. If we keep it like that, we'd need to document that cancelling a poll throws an error :/

Reply:
So, to be very clear. It doesn't `emit` nor does it `throw` an error when a `LIBUSB_TRANSFER_CANCELLED` is encountered. With the above changes though, it would now pass it back in the call back upon the finish of the `startPoll`.

This is good, as in prior to this code, one would never know this error even occurred at all.

Reply:
It is common practice to check for an error in callback and assume failure, though. If the function was [promisified](https://nodejs.org/api/util.html#utilpromisifyoriginal) I believe it would result in an error being thrown.

Reply:
Agreed. It is common, but not necessarily the only way to handle the situation depending on the error. Error's are meant to be fatal/unrecoverable in those situations. As far as I can tell the `LIBUSB_TRANSFER_CANCELLED` isn't necessarily and `error` in the sense that we are talking about at all.

This is all my understanding of the libusb stack and how it's being utilized here, so I could be way off map here.

I still think you're correct in stating we need to document it, I just wanted to be clear it isn't a literal `throw` or `emit`. Personally I'd argue that this should actually be a `boolean` getting sent back? Though this is also a large deviation from the current usage pattern.

Reply:
👍 I like this proposal.  Although it changes the signature, adding an extra parameter doesn't break backwards compatibility.

I assume in the case of `LIBUSB_TRANSFER_CANCELLED`,  the error would be undefined and we'd set the cancelled boolean?

Reply:
Got it, I can make these changes likely later this week. Bluepill arrived as well but I haven't had any time to toy around with it or the testing firmware repo.

Reply:
I've pushed code for this, obviously adding a test to the firmware would be best. Though this will take more cycles, which I'm planning, but don't have at the moment. Feel free to just ignore the PR until then if you'd like. I can always add test cases via another PR later as well.